### PR TITLE
Rename MultiscaleChunkedImage to MultiscaleSpatialImage and use Maps with dimension keys

### PR DESCRIPTION
**Pull request by PaulHax**
related #478

### CODE REVIEW

File: src/IO/MultiscaleSpatialImage.js [hunk 1]
```diff
@@ -13,15 +14,15 @@ const haveSharedArrayBuffer = typeof window.SharedArrayBuffer === 'function'
...
   scaleInfo = [{
     // scale 0 information
     dims: ['x', 'y'], // Valid elements: 'c', 'x', 'y', 'z', or 't'
     coords: .get() Promise resolves a Map('x': Float64Array([0.0, 2.0, ...), 'y' ...
-    numberOfCXYZTChunks: [1, 10, 10, 5, 1], // array shape in chunks
-    sizeCXYZTChunks: [1, 64, 64, 64, 1], // chunk shape in elements
-    sizeCXYZTElements: [1, 1, 1, 1, 1], // array shape in elements
+    chunkCount: [1, 10, 10, 5, 1], // array shape in chunks
+    chunkSize: [1, 64, 64, 64, 1], // chunk shape in elements
+    arrayShape: [1, 1, 1, 1, 1], // array shape in elements
```
Comment (thewtex) [inline src/IO/MultiscaleSpatialImage.js:25]: Update example values

File: src/IO/ZarrMultiscaleSpatialImage.js [hunk 2]
```diff
@@ -25,8 +26,31 @@ const dtypeToComponentType = new Map([
...
+
+  return {
+    get(dim) {
+      if (coords.get(dim) === null) {
+        // make array
+        const dimIdx = dims.indexOf(dim)
+        const spacing = imageScale[dimIdx] * datasetScale[dimIdx]
+        const coordsPerElement = new Float32Array(shape[dimIdx])
+        for (let i = 0; i < coordsPerElement.length; i++) {
+          coordsPerElement[i] = i * spacing // + origin translate transformations not implemented
```
Comment (thewtex) [inline src/IO/ZarrMultiscaleSpatialImage.js:43]: Let's go ahead and add translation  / origin support now please

**Review by thewtex — COMMENTED** [review]
image

A few minor issues inline

File: src/IO/MultiscaleSpatialImage.js (hunk 1 above)
Comment (PaulHax) [reply src/IO/MultiscaleSpatialImage.js]: Done.

**Review by thewtex — APPROVED** [review]

**Pull request merged**